podvm/files/etc/kata-opa/allow-all.rego

```diff
@@ -0,0 +1,39 @@
+package agent_policy
+
+default AddARPNeighborsRequest := true
```

Comment:
What if there is no default configure here. is an agent function enabled or disabled by default?

Reply:
OPA returns an empty response in that case, and policy.rs converts that response automatically to false (denied).

Typically, it's better to specify default values (even if they are false) because:
a. That kind of Policy is easier to read/understand.
b. The policy.rs log text is easier to read/understand.

Sharing my latest understanding on OPA default values, just in case it's useful for you guys too:

**Example1**

- No default value for AddARPNeighborsRequest, and
- AddARPNeighborsRequest rules similar to:

AddARPNeighborsRequest {
    **false**
}

Querying OPA for AddARPNeighborsRequest returns an **empty value** (a.k.a. the default value) - due to the rules being evaluated to false.

**Example2**

- No default value for AddARPNeighborsRequest, and

AddARPNeighborsRequest {
    **true**
}

Querying OPA for AddARPNeighborsRequest returns **true** - due to the rules being evaluated to true.

**Example3**

default AddARPNeighborsRequest := **true**

AddARPNeighborsRequest {
    **false**
}

Querying OPA for AddARPNeighborsRequest returns **true** (the default value) - due to the rules being evaluated to false.

**Example4**

default AddARPNeighborsRequest := **false**

AddARPNeighborsRequest {
    **false**
}

Querying OPA for AddARPNeighborsRequest returns **false** (the default value) - due to the rules being evaluated to false.

**Example4**

default AddARPNeighborsRequest := **false**

AddARPNeighborsRequest {
    **true**
}

Querying OPA for AddARPNeighborsRequest returns **true** - due to the rules being evaluated to true.

Reply:
Thanks @danmihai1 . This is very helpful

Reply:
Ah, that's good. if they evaluate to false by default, we don't have to worry too much about new kata endpoints and policies that don't cover those.